### PR TITLE
logging: Update to new logging API

### PR DESCRIPTION
**Pull request by ajtowns**
Updates various logging calls to the new API from #28318. All commits are scripted diffs, so should be easy to update if needed, and also easy to reuse the scripts to update other code if needed when rebasing it after this is merged. Changes all uses of `LogPrintLevel()` where the level is hardcoded, and changes all `LogPrintf` and `LogPrint` uses in init.cpp.

### CODE REVIEW

**Comment by ajtowns** [comment]
Depending on how many conflicts drahtbot collects, maybe this should be targeting a merge around 27.0's feature freeze?

Many of the log statements in init.cpp could be done at a better level, eg `LogPrintf("Error: Out of memory. Terminating.\n");` should obviously be logging at the "error" level, not the "info" level. In order to keep this PR scripted-diff only, I won't fix that here; but it can fixed in an independent PR either before or after this PR is merged.

**Comment by fanquake** [comment]
Concept ACK - only 3 conflicts, and none looks like priority projects.

**Review by stickies-v — APPROVED** [review]
ACK 2632d038754d975e76926b7c068309c7aadc82f5

Since we don't have significant merge conflicts I would this support this PR having a non-scripted diff commit to first fix incorrect levels (instead of having to fix and then re-fix) but am okay with this approach too.

Suggested fixes in https://github.com/bitcoin/bitcoin/compare/master...stickies-v:bitcoin:2024-01/fix-log-level-29231 - happy to open a separate pull for it too?

**Comment by ajtowns** [comment]
> Suggested fixes in [master...stickies-v:bitcoin:2024-01/fix-log-level-29231](https://github.com/bitcoin/bitcoin/compare/master...stickies-v:bitcoin:2024-01/fix-log-level-29231) - happy to open a separate pull for it too?

Added these commits

**Review by stickies-v — APPROVED** [review]
ACK 1441ab74d6cd40e75f1dba80b68bc4b1791bf667

**Comment by ajtowns** [comment]
Rebased

**Comment by epiccurious** [comment]
> Changes all uses of LogPrintLevel() where the level is hardcoded, and changes all LogPrintf and LogPrint uses in init.cpp.

Trying to understand the concept here. What problem does this solve?

**Comment by stickies-v** [comment]
re-ACK 8d46bd6750b47caaf2556f915800c6a22c997bc3 - no change since 1441ab74d6cd40e75f1dba80b68bc4b1791bf667 except for addressing merge conflicts

File: src/init.cpp [hunk 1]
```diff
@@ -1138,11 +1138,11 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         return false;
     }
 
-    LogPrintf("Using at most %i automatic connections (%i file descriptors available)\n", nMaxConnections, nFD);
+    LogInfo("Using at most %i automatic connections (%i file descriptors available)\n", nMaxConnections, nFD);
 
     // Warn about relative -datadir path.
     if (args.IsArgSet("-datadir") && !args.GetPathArg("-datadir").is_absolute()) {
-        LogPrintf("Warning: relative datadir option '%s' specified, which will be interpreted relative to the "
+        LogWarning("Warning: relative datadir option '%s' specified, which will be interpreted relative to the "
```
Comment (davidgumberg) [inline src/init.cpp:1145]:
```suggestion
        LogWarning("Relative datadir option '%s' specified, which will be interpreted relative to the "
```

**Comment by davidgumberg** [comment]
utACK https://github.com/bitcoin/bitcoin/commit/8d46bd6750b47caaf2556f915800c6a22c997bc3

Probably tangential to this PR:
 Is `LogPrintLevel` in `void libevent_log_cb(...)` an example of:

> "when [...] a category needs to be added for an info/warning/error log message"

<details>

<summary>Otherwise:</summary>

```diff
 static void libevent_log_cb(int severity, const char *msg)
 {
-    BCLog::Level level;
     switch (severity) {
     case EVENT_LOG_DEBUG:
-        level = BCLog::Level::Debug;
+        LogDebug(BCLog::LIBEVENT, "%s\n", msg);
         break;
     case EVENT_LOG_MSG:
-        level = BCLog::Level::Info;
+        LogInfo("%s\n", msg);
         break;
     case EVENT_LOG_WARN:
-        level = BCLog::Level::Warning;
+        LogWarning("%s\n", msg);
         break;
     default: // EVENT_LOG_ERR and others are mapped to error
-        level = BCLog::Level::Error;
+        LogError("%s\n", msg);
         break;
     }
-    LogPrintLevel(BCLog::LIBEVENT, level, "%s\n", msg);
 }
```

**Comment by ajtowns** [comment]
Rebased over #28834, removed redundancy from `LogWarning("Warning: ..")`

**Comment by laanwj** [comment]
@ajtowns Sorry, i didn't mean to derail this, i shouldn't have commented.

**Pull request closed without merging**